doc/source/io.rst

```diff
@@ -3884,28 +3885,54 @@ destination DataFrame as well as a preferred column order as follows:
                              index_col='index_column_name',
                              col_order=['col1', 'col2', 'col3'], project_id = projectid)
 
-Finally, you can append data to a BigQuery table from a pandas DataFrame
-using the :func:`~pandas.io.to_gbq` function. This function uses the
-Google streaming API which requires that your destination table exists in
-BigQuery. Given the BigQuery table already exists, your DataFrame should
-match the destination table in column order, structure, and data types.
-DataFrame indexes are not supported. By default, rows are streamed to
-BigQuery in chunks of 10,000 rows, but you can pass other chuck values
-via the ``chunksize`` argument. You can also see the progess of your
-post via the ``verbose`` flag which defaults to ``True``. The http
-response code of Google BigQuery can be successful (200) even if the
-append failed. For this reason, if there is a failure to append to the
-table, the complete error response from BigQuery is returned which
-can be quite long given it provides a status for each row. You may want
-to start with smaller chunks to test that the size and types of your
-dataframe match your destination table to make debugging simpler.
+By default, the query result is returned as a dataframe, but you can redirect the results
+to a destination table via the ``destination_table`` argument. Use the ``if_exists`` argument
+to specify the action to take when the destination table already exists. For example, the following
+query will insert the results of the query directly to a bigquery table, instead of returning a dataframe:
+
+.. code-block:: python
+
+   data_frame = pd.read_gbq('SELECT * FROM test_dataset.test_table',
+                             index_col='index_column_name',
+                             col_order=['col1', 'col2', 'col3'], project_id = projectid,
```

Comment:
I have prepared 2 proposals in order to allow the user to choose to upload a Pandas Dataframe to a BigQuery table or run a SQL-Like query and have the results goto a BigQuery table. I would prefer proposal 1. I will create a new PR once the decision is made.

######Proposal 1
 
Modify ``pd.to_gbq()`` to support both ``dataframe`` and ``query`` as arguments.  Both will be optional, but at least 1 is required. In the new revision the ``dataframe`` parameter will be changed to optional so that users can either specify to upload either a pandas DataFrame or run SQL-Like Query and have the results sent directly to a BigQuery table.

function name: ``pd.to_gbq()``
``` Arguments
    ----------
   dataframe : DataFrame (optional)
        DataFrame to be written. Note: one of dataframe or query is required.
    query : str (optional)
        SQL-Like Query to return data values. Note: one of dataframe or query is required.
    destination_table : string 
        Name of table to be written, in the form 'dataset.tablename'
    project_id : str
        Google BigQuery Account project ID.
    chunksize : int (default 10000)
        Number of rows to be inserted in each chunk from the dataframe.
    verbose : boolean (default True)
        Show percentage complete
    reauth : boolean (default False)
        Force Google BigQuery to reauthenticate the user. This is useful
        if multiple accounts are used.
    if_exists : {'fail', 'replace', 'append'}, default 'fail'
            - fail: If table exists, do nothing.
            - replace: If table exists, drop it, recreate it, and insert data.
            - append: If table exists, insert data. Create if does not exist.
    """

    # only one will survive
    if dataframe and query:
        raise AssertionError('Only one of dataframe and query can be provided')

   # at least one of dataframe and query is required
      if not (dataframe or query):
         raise AssertionError('At least one of dataframe and query can be provided')
``` 


######Proposal 2
- ``pd.to_gbq()`` will have still ``dataframe`` as a required parameter (i.e. no change )
- create a new ``pd`` function to support running a SQL-Like Query and having the results go directly to a BigQuery destination table. This can be used for running large queries which require the BigQuery allowLargeResults parameter.
Proposed function name :   ``pd.to_gbq_large``